tests/phpunit/class-test-core-sitemaps.php

```diff
@@ -159,5 +216,237 @@ public function test_core_sitemaps_xml() {
+				'loc'     => 'http://' . WP_TESTS_DOMAIN . '/?sitemap=users&paged=1',
+				'lastmod' => '',
+			),
+		);
+
+		$this->assertSame( $expected, $entries );
+	}
+
+	/**
+	 * Test default sitemap entries with permalinks on.
+	 */
+	public function test_get_sitemap_entries_post_with_permalinks() {
+		$this->set_permalink_structure( '/%year%/%postname%/' );
+
+		$entries = $this->_get_sitemap_entries();
+
+		$expected = array(
+			array(
+				'loc'     => 'http://' . WP_TESTS_DOMAIN . '/sitemap-posts-post-1.xml',
+				'lastmod' => '',
+			),
+			array(
+				'loc'     => 'http://' . WP_TESTS_DOMAIN . '/sitemap-posts-page-1.xml',
+				'lastmod' => '',
+			),
+			array(
+				'loc'     => 'http://' . WP_TESTS_DOMAIN . '/sitemap-taxonomies-category-1.xml',
+				'lastmod' => '',
+			),
+			array(
+				'loc'     => 'http://' . WP_TESTS_DOMAIN . '/sitemap-taxonomies-post_tag-1.xml',
+				'lastmod' => '',
+			),
+			array(
+				'loc'     => 'http://' . WP_TESTS_DOMAIN . '/sitemap-users-1.xml',
+				'lastmod' => '',
+			),
+		);
+
+		$this->assertSame( $expected, $entries );
+	}
+
+	/**
+	 * Test sitemap index entries with public and private custom post types.
+	 */
+	public function test_get_sitemap_entries_custom_post_types() {
+		// Register and create a public post type post.
+		register_post_type( 'public_cpt', array( 'public' => true ) );
+		$this->factory->post->create( array( 'post_type' => 'public_cpt' ) );
+
+		// Register and create a private post type post.
+		register_post_type( 'private_cpt', array( 'public' => false ) );
+		$this->factory->post->create( array( 'post_type' => 'private_cpt' ) );
+
+		$entries = wp_list_pluck( $this->_get_sitemap_entries(), 'loc' );
+
+		$this->assertTrue( in_array( 'http://' . WP_TESTS_DOMAIN . '/sitemap-posts-public_cpt-1.xml', $entries, true ), 'Public CPTs are not in the index.' );
+		$this->assertFalse( in_array( 'http://' . WP_TESTS_DOMAIN . '/sitemap-posts-private_cpt-1.xml', $entries, true ), 'Private CPTs are visible in the index.' );
+
+		// Clean up.
+		unregister_post_type( 'public_cpt' );
+		unregister_post_type( 'private_cpt' );
+	}
+
+	/**
+	 * Tests getting a URL list for post type post.
+	 */
+	public function test_get_url_list_post() {
+		$providers = core_sitemaps_get_sitemaps();
+
+		$post_list = $providers['posts']->get_url_list( 1, 'post' );
+
+		$expected = $this->_get_expected_url_list( 'post', self::$posts );
+
+		$this->assertEquals( $expected, $post_list );
+	}
+
+	/**
+	 * Tests getting a URL list for post type page.
+	 */
+	public function test_get_url_list_page() {
+		// Short circuit the show on front option.
+		add_filter( 'pre_option_show_on_front', '__return_true' );
+
+		$providers = core_sitemaps_get_sitemaps();
+
+		$post_list = $providers['posts']->get_url_list( 1, 'page' );
+
+		$expected = $this->_get_expected_url_list( 'page', self::$pages );
+
+		$this->assertEquals( $expected, $post_list );
+
+		// Clean up.
+		remove_filter( 'pre_option_show_on_front', '__return_true' );
```

Comment:
Cleanup should usually happen before the assertion. Otherwise, if the assertion fails, this code is not reached.